In [2]:
print('Hello')

Hello


# Getting start with Jupiter Python
## We have done
* Install[Python](https://www.python.org/downloads/) 3.12
* Install Jupyter Python in PyCharm

# Raw NBConvert
45 + 49

In [3]:
i = 5
type(i)

int

In [4]:
f = 3.12
type(f)

float

In [5]:
b = True
type(b)

bool

In [6]:
s = 'String'
print(s)

String


In [7]:
d = {'foo' : 1, 'bar' : 2.3, 's' : 'my first dict'}
print(d)

{'foo': 1, 'bar': 2.3, 's': 'my first dict'}


In [13]:
import math
a = math.pi
b = math.pow(4, 8)
c = math.sqrt(b)
print(f' {a}  = a  {b}  =b   {c}   =c ')

 3.141592653589793  = a  65536.0  =b   256.0   =c 


In [17]:
i = 5
n = False
print('Our first value is %s. Our second value equals %s.' %(i, n))

Our first value is 5. Our second value equals False.


In [19]:
a = 6
b = 12.5
if a == 6 and f > 15:
    print("a = 6 and f > 15")
elif a== 6 and f < 15:
    print('a = 6 and f < 15')
else: print(' a = ? and b = ?')

a = 6 and f < 15


In [22]:
ar = [1, 2, 3]
print(ar)
for e in ar:
    print(e)
    break

[1, 2, 3]
1


In [26]:
import time
for n in range(12):
    time.sleep(1)
    print(f'Number is {n}') 


Number is 0
Number is 1
Number is 2
Number is 3
Number is 4
Number is 5
Number is 6
Number is 7
Number is 8
Number is 9
Number is 10
Number is 11


In [27]:
i = 6
while i < 10:
    print(i)
    i += 1

6
7
8
9


In [30]:
def add(x):
    y = x + 1
    return y


In [31]:
sum = lambda x: x+x
summa = sum(5)
print(summa)

10


<h1> Heading 1 </h1>
<h2> Heading 2 </h2>


\begin{equation*}
\left(sum_{k=1}^1 a_k b_k\right)^2
\end{equation*}

In [34]:
!ipconfig


Ќ бва®©Є  Їа®в®Є®«  IP ¤«п Windows


Ђ¤ ЇвҐа Ethernet Ethernet 2:

   ‘®бв®п­ЁҐ б।л. . . . . . . . : ‘।  ЇҐаҐ¤ зЁ ­Ґ¤®бвгЇ­ .
   DNS-бгддЁЄб Ї®¤Є«о祭Ёп . . . . . : 

Ђ¤ ЇвҐа Ethernet Ethernet 3:

   DNS-бгддЁЄб Ї®¤Є«о祭Ёп . . . . . : 
   ‹®Є «м­л© IPv6- ¤аҐб Є ­ «  . . . : fe80::afb1:e125:2e2:49b%17
   IPv4- ¤аҐб. . . . . . . . . . . . : 192.168.56.1
   Њ бЄ  Ї®¤бҐвЁ . . . . . . . . . . : 255.255.255.0
   Ћб­®ў­®© и«о§. . . . . . . . . : 

Ђ¤ ЇвҐа ЎҐбЇа®ў®¤­®© «®Є «м­®© бҐвЁ ЃҐбЇа®ў®¤­ п бҐвм:

   DNS-бгддЁЄб Ї®¤Є«о祭Ёп . . . . . : 
   IPv4- ¤аҐб. . . . . . . . . . . . : 192.168.64.157
   Њ бЄ  Ї®¤бҐвЁ . . . . . . . . . . : 255.255.255.0
   Ћб­®ў­®© и«о§. . . . . . . . . : 192.168.64.14


In [36]:
#x = lambda f, t, x: f = ((9 * math.pi * t) + (10 * math.cos(x))) / (math.sqrt(t) - math.sin(x)) * math.pow(math.e, -2t)

def x(f,t,x):
    f = ((9 * math.pi * t) + (10 * math.cos(x))) / (math.sqrt(t) - math.sin(x)) * math.pow(math.e, -2 * t)
    return f


In [2]:
def find_numbers_in_interval(a, b, num1, num2, num3):
   
    result = []
    if a <= num1 <= b:
        result.append(num1)
    if a <= num2 <= b:
        result.append(num2)
    if a <= num3 <= b:
        result.append(num3)
    
    return result


a = 10
b = 20
num1 = 15
num2 = 25
num3 = 12

result = find_numbers_in_interval(a, b, num1, num2, num3)
print("Числа, принадлежащие интервалу:", result)


Числа, принадлежащие интервалу: [15, 12]


In [4]:
import random

def random_numbers_generator():
    numbers = []
    for _ in range(20):
        numbers.append(random.randint(0, 10))
    return numbers

# Пример использования:
numbers = random_numbers_generator()
for number in numbers:
    print(number)


0
4
3
8
10
10
6
1
3
4
1
4
5
0
5
10
1
7
8
3


In [6]:
def calculate_sum_of_sequence(n):
    total_sum = 0
    for i in range(1, n + 1):
        total_sum += i / (i + 1 + 2)
    return total_sum


n = 20
result = calculate_sum_of_sequence(n)
print(f"Сумма {n} элементов последовательности: {result}")


Сумма 20 элементов последовательности: 14.297125466739482


1.	Почему для работы с большими данными и для машинного обучения выбирают язык Python? Перечислите не менее трех причин.
Ответ:
    Python является популярным выбором для работы с большими данными и машинного обучения по следующим причинам:

Богатая экосистема библиотек:

Python предлагает широкий набор библиотек, специально разработанных для работы с данными и машинным обучением, таких как:
NumPy и Pandas для обработки и анализа данных.
TensorFlow, PyTorch, Scikit-learn для машинного обучения и глубокого обучения.
Matplotlib и Seaborn для визуализации данных.
Простота и читаемость кода:

Python имеет простой синтаксис, который делает его доступным даже для начинающих разработчиков. Это позволяет быстро писать и понимать код, что особенно важно в сложных проектах с большими объемами данных.
Сообщество и поддержка:

Python имеет огромное и активное сообщество разработчиков и исследователей, которые создают новые инструменты и делятся решениями. Это обеспечивает быстрый доступ к ресурсам, поддержке и актуальной документации.

2.	В чем особенности среды Jupyter Notebook, почему ее выбирают специалисты по Data Science? 
Ответ:
Jupyter Notebook популярен среди специалистов по Data Science благодаря возможности интерактивного выполнения кода и мгновенной визуализации результатов. Он поддерживает текстовые и кодовые блоки, что упрощает документирование и объяснение данных прямо в рабочем процессе. Jupyter также гибок, поддерживает различные языки программирования, и его легко использовать для экспериментов и обмена результатами.
3. Перечислите основные библиотеки Python для машинного обучения.
Ответ:
Numpy, Matplotlib, scikit-learn
4.	Какие типы данных Python вы знаете?
Ответ:
int — целые числа (например, 5, -10)
float — числа с плавающей точкой (например, 3.14, -2.5)
str — строки (например, "Hello", 'Python')
bool — логические значения (например, True, False)
list — списки (например, [1, 2, 3], ['a', 'b', 'c'])
tuple — кортежи (например, (1, 2, 3), ('x', 'y', 'z'))
dict — словари (например, {'key': 'value'}, {1: 'one', 2: 'two'})
5.	Назовите основные языковые конструкции Python.
Ответ:
Условные операторы: if, elif, else — для выполнения кода в зависимости от условий.
Циклы: for, while — для повторного выполнения блока кода.
Функции: def — для определения функций, которые могут быть вызваны повторно.
Классы: class — для создания объектов и организации кода в объектно-ориентированном стиле.
Операторы управления потоком: break, continue, return — для управления выполнением циклов и функций.
Обработчик исключений: try, except, finally — для обработки ошибок в программе.
6.	Какое назначение у оператора if … else?
Ответ:
Оператор if … else в Python используется для выполнения разных блоков кода в зависимости от того, истинно или ложно заданное условие.

if: Проверяет условие. Если условие истинно, выполняется соответствующий блок кода.
else: Выполняется, если условие в if ложно. Это альтернатива, которая предоставляет код, выполняемый в противном случае.
7.	Когда заканчивается выполнение оператора while?
Оператор while в Python завершает своё выполнение, когда условие, указанное в цикле, становится ложным.
8. Какие способы задания функций есть в Python?
Ответ:
В Python есть несколько способов задания функций:

1. **Обычная функция**:
   Используется для определения стандартных функций с помощью ключевого слова `def`.
   ```python
   def my_function(x):
       return x * 2
   ```

2. **Анонимная функция (lambda)**:
   Позволяет создавать функции без имени, обычно для кратковременного использования.
   ```python
   square = lambda x: x * x
   ```

3. **Функция-генератор**:
   Использует `yield` для создания генераторов, которые могут возвращать несколько значений по одному.
   ```python
   def count_up_to(max):
       count = 1
       while count <= max:
           yield count
           count += 1
   ```

4. **Функция как объект (использование функций в качестве параметров)**:
   Функции могут передаваться в качестве параметров другим функциям, что позволяет создавать высокоуровневые абстракции и декораторы.
   ```python
   def apply_function(func, value):
       return func(value)
   
   result = apply_function(lambda x: x + 1, 5)
   ```

5. **Методы класса (функции внутри классов)**:
   Определяются внутри классов и работают с экземплярами объектов этого класса.
   ```python
   class MyClass:
       def method(self, x):
           return x * 3
   ```

Эти способы позволяют использовать функции в различных контекстах, от простых вычислений до сложных структур программирования.
9.	Что такое Лямбда-функция?
Ответ:
Лямбда-функция в Python — это небольшая анонимная функция, определяемая с помощью ключевого слова lambda. Она используется для создания функций на лету, когда требуется короткая и простая функция, обычно для временного использования. Лямбда-функции имеют ограничение на одну выражение и не могут содержать сложные конструкции, такие как инструкции if, for, while.
10.	Какая библиотека Python отвечает за выполнение основным математических функций?
Ответ:
В Python основная библиотека для выполнения математических функций — это math.
11.	Назовите дополнительные возможности использования среды Jupyter Notebook.
Ответ:

Среда Jupyter Notebook предлагает множество дополнительных возможностей, помимо стандартного выполнения кода и визуализации данных:

Создание интерактивных визуализаций:

Использование библиотек, таких как Plotly или Bokeh, для создания динамических и интерактивных графиков и визуализаций.
Документирование и аннотирование:

Вставка текстовых ячеек с использованием Markdown и LaTeX для добавления пояснений, формул и документации прямо в ноутбук.
Интерактивные виджеты:

Использование ipywidgets для создания интерактивных элементов, таких как слайдеры и кнопки, которые позволяют взаимодействовать с данными и моделями.
Запуск кода на разных языках программирования:

Поддержка многих языков программирования (через ядра), таких как R, Julia, Scala и другие, что позволяет работать с разными инструментами и языками в одном окружении.
Совместная работа и обмен:

Возможность совместного редактирования и обмена ноутбуками через платформы, такие как JupyterHub или Google Colab.
Использование расширений:

Установка и использование расширений, таких как nbextensions, для добавления дополнительного функционала и улучшения интерфейса.
Экспорт в различные форматы:

Возможность экспорта ноутбуков в форматы, такие как HTML, PDF, Markdown, для удобного представления и публикации результатов.
Автоматическое тестирование и отладка:

Использование инструментов и библиотек для автоматического тестирования кода и отладки, таких как pytest и debugging tools.
Эти возможности делают Jupyter Notebook мощным инструментом для анализа данных, создания отчетов и проведения интерактивных исследований.